internal/pkg/manifest/pipeline.go

```diff
@@ -85,56 +91,37 @@ type Source struct {
 	Properties   map[string]interface{} `yaml:"properties"`
 }
 
-// PipelineStage represents configuration for each deployment stage
-// of a workspace. A stage consists of the Archer Environment the pipeline
-// is deloying to and the containerized applications that will be deployed.
-type PipelineStage struct {
-	*AssociatedEnvironment `yaml:",inline"`
-	Applications           []*archer.Application `yaml:"-"`
-}
-
-// AssociatedEnvironment defines the necessary information a pipline stage
-// needs for an Archer Environment.
-type AssociatedEnvironment struct {
-	Project   string `yaml:"-"`    // Name of the project this environment belongs to.
-	Name      string `yaml:"name"` // Name of the environment, must be unique within a project.
-	Region    string `yaml:"-"`    // Name of the region this environment is stored in.
-	AccountID string `yaml:"-"`    // Account ID of the account this environment is stored in.
-	Prod      bool   `yaml:"-"`    // Whether or not this environment is a production environment.
-}
-
 // CreatePipeline returns a pipeline manifest object.
-func CreatePipeline(provider Provider, stages ...PipelineStage) (archer.Manifest, error) {
+func CreatePipeline(pipelineName string, provider Provider, stages ...PipelineStage) (*PipelineManifest, error) {
 	// TODO: #221 Do more validations
-	var defaultStages []PipelineStage
 	if len(stages) == 0 {
-		defaultStages = []PipelineStage{
-			{
-				AssociatedEnvironment: &AssociatedEnvironment{
-					Name: "test",
-				},
-			},
-			{
-				AssociatedEnvironment: &AssociatedEnvironment{
-					Name: "prod",
-				},
-			},
+		return nil, fmt.Errorf("a pipeline %s can not be created without a deployment stage",
+			pipelineName)
+	}
+
+	var projectName = stages[0].ProjectName
```

Comment:
The first stage is only used to pull out `ProjectName`? Can you help me understand what's going on here exactly?